packages/terra-clinical-data-grid/docs/README.md

```diff
@@ -0,0 +1,93 @@
+
+### Rows
+
+Rows define the cells rendered within the row as well as the row's selection properties.
+
+Rows are provided to sections to be rendered by the DataGrid. The order in which the rows are provided will be the order in which they are rendered.
+
+|Name|Type|Is Required|Default Value|Description|
+|---|---|---|---|---|
+|**id**|string|required|none|An identifier for the row. This identifier should be unique across the set of rows provided to a given section.|
+|**isSelectable**|bool|optional|none|A boolean indicating whether or not the row is selectable.|
+|**isSelected**|bool|optional|none|A boolean indicating whether or not the row should render as selected.|
+|**cells**|array|optional|[]|An array of cell objects that define the content to be rendered in the row. The order in which the cells are rendered is determined by the order of the `pinnedColumns` and `overflowColumns` props.|
+
+### Cells
+
+Cells define the content rendered in a given section, row, and column.
+
+|Name|Type|Is Required|Default Value|Description|
+|---|---|---|---|---|
+|**columnId**|string|required|none|A column identifier for the cell. This identifier should be unique across the set of cells provided to a given row.|
+|**isSelectable**|bool|optional|none|A boolean indicating whether or not the cell is selectable.|
+|**isSelected**|bool|optional|none|A boolean indicating whether or not the cell should render as selected.|
+|**component**|component|optional|none|A component to render within the cell.|
+
+#### Creating components for Cells
+
+The DataGrid will provide a explicitly sized container within which the Cell `component` will be rendered. A height and width of `100%` should be set on the component to ensure it fills the available area.
+
+## Creating Custom Content
+
+The various DataGrid containers (column header, section header, cell) support the presentation of custom components. However, there are a few rules that must successful component implementation must adhere to.
+
+The DataGrid provides a number of styles for mouse/keyboard interaction states (click, hover, selection). Custom components should generally render with a transparent background to ensure those default styles bleed through the component.
+
+The DataGrid will provide a explicitly sized container within which the component will rendered. A height and width of `100%` should be set on the component to ensure it fills the area provided to it.
+
+If there are any mouse/keyboard interactive elements within the component, and the component is rendered inside a selectable DataGrid container (column header, section header, cell), events may propagate in ways that the component is not expecting. If necessary, the events originating from those interactive elements could not be propagated to prevent undesired DataGrid interactions.
```

Comment:
```diff
- If necessary, the events originating from those interactive elements could not be propagated to prevent undesired DataGrid interactions.
+ If necessary, propigation should be prevented for the events originating from those interactive elements  to avoid undesired DataGrid interactions.
```